webui/react/src/pages/JobQueue/JobQueue.table.tsx

```diff
@@ -0,0 +1,126 @@
+import Icon from 'components/Icon';
+import Link from 'components/Link';
+import { relativeTimeRenderer } from 'components/Table';
+import { paths } from 'routes/utils';
+import { Determinedjobv1Type } from 'services/api-ts-sdk';
+import { Job } from 'types';
+import { truncate } from 'utils/string';
+import { jobTypeIconName, jobTypeLabel } from 'utils/types';
+
+type Renderer<T> = (_: unknown, record: T) => ReactNode;
+export type JobTypeRenderer = Renderer<Job>;
+
+export const SCHEDULING_VAL_KEY = 'schedulingVal';
+
+// translate job type to paths from routes utils
+const jobTypeToPath = (type: Determinedjobv1Type, id: string): string => {
+  switch (type) {
+    case Determinedjobv1Type.EXPERIMENT:
+      return paths.experimentDetails(id);
+    case Determinedjobv1Type.NOTEBOOK:
+    case Determinedjobv1Type.TENSORBOARD:
+      // FIXME we need the command's service address
+      return paths.taskList();
+    default:
+      return '';
+  }
+};
+
+export const columns: ColumnType<Job>[] = [
+  { key: 'jobsAhead' },
+  {
+    dataIndex: 'jobId',
+    key: 'jobId',
+    render: (_: unknown, record: Job): ReactNode => {
+      const pagePath = jobTypeToPath(record.type, record.entityId);
+      const label = truncate(record.jobId, 6, '');
+      if (pagePath) {
+        return <Link path={pagePath}>{label}</Link>;
+      }
+      return label;
+    },
+    title: 'ID',
+  },
+  {
+    dataIndex: 'type',
+    key: 'type',
+    render: (_: unknown, record: Job): ReactNode => {
+      const title = jobTypeLabel(record.type);
+      const TypeCell = <Tooltip placement="topLeft" title={title}>
+        <div className={''}>
+          <Icon name={jobTypeIconName(record.type)} />
+        </div>
+      </Tooltip>;
+      return TypeCell;
+    },
+    title: 'Type',
+  },
+  {
+    key: 'name',
+    render: (_: unknown, record: Job): ReactNode => {
+      const pagePath = jobTypeToPath(record.type, record.entityId);
+      const label = record.name ?
+        record.name : <span>{jobTypeLabel(record.type)} {truncate(record.entityId, 6, '')}</span>;
+      if (pagePath) {
+        return <Link path={pagePath}>{label}</Link>;
+      }
+      return label;
+    },
+    title: 'Job Name',
+  },
+  {
+    dataIndex: 'priority',
+    key: SCHEDULING_VAL_KEY,
+    title: 'Priority', // TODO or weight? in fairshare
+  },
+  {
+    dataIndex: 'submissionTime',
+    key: 'submitted',
+    render: (_: unknown, record: Job): ReactNode =>
+      record.submissionTime && relativeTimeRenderer(record.submissionTime),
+    title: 'Submitted',
+  },
+  {
+    key: 'slots',
+    render: (_: unknown, record: Job): ReactNode => {
+      const cell = <span>
+        {record.allocatedSlots} / {record.requestedSlots}
+      </span>;
+      return cell;
+    },
+    title: 'Slots (Scheduled/Queued)',
+  },
+  {
+    key: 'state',
+    render: (_: unknown, record: Job): ReactNode => {
+      return <>
+        <Badge state={record.summary.state} type={BadgeType.State} />
```

Comment:
**chore:** just nothing that this threw an error of `summary` may be undefined for me.
```
record.summary?.state
```

Reply:
👍🏼 I still haven't decided if there is a usecase we want the api to be able to advertise jobs without a summary availalbe. 🤔  depending on that we'd change how we handle this here. It shouldn't throw an error in either case.